src/Pages/LeaderBoard.js

```diff
@@ -1,39 +1,112 @@
-import React, { useEffect, useState } from 'react';
-import { useNavigate } from "react-router-dom";
-import data from "./config.contracts.json"
+import React, { useEffect, useState } from 'react'
 
-var dataArray = []
-var VaRofContracts = ["$$$","$$$","$$$", "$$$", "$$$", "$$$"]
+const APIURL = "https://api.thegraph.com/subgraphs/name/greg-torrington/greg-v3"
+const assetQuery = `
+query {
+  assets {
+    id
+    name
+  	count
+  }
+}
+`
 
-var navigate;
+function LeaderBoard(props) {
+  const [searchTerm, setSearchTerm] = useState("")
+  const [loading, setLoading] = useState(true)
+  const [totalVaR, setTotalVaR] = useState(0)
+  const [allProtocols, setAllProtocols] = useState([])
 
-export var chosenData
+  async function fetchBlockChainData() {
+    let protocols = []
+    let protocolInfo = []
 
-async function sortData(){
+    const assetResponse = await props.client.query(assetQuery).toPromise()
+    let rawAssetData = assetResponse.data.assets
+    protocolInfo[0] = "Maker"
+    protocolInfo[1] = rawAssetData
 
-  data.map( (data) => {
-      dataArray.push(data)
-  })
+    let usersPromiseArray = []
+    let protocolVaR = 0;
+    for (var i=0; i<protocolInfo[1].length; i++){
+        var noUsers = protocolInfo[1][i].count
+  
+        let skip = 0
+        let first = 1000
+        while (noUsers>0) {
+            if (noUsers<1000){first=noUsers}
 
-}
+            let usersQuery = `
+                query {
+                    users (
+                    first: `+first+`
+                    after: `+skip+`
+                    ){
+                        id
+                        balance
+                        allowance
+                    }
+                }
+                `
 
-async function navigateToProtolPage(i){
-  chosenData = dataArray[i]
-  navigate("/protocol");
-}
+            skip += 1000
+            noUsers = noUsers - 1000
 
-function LeaderBoard() {
+            usersPromiseArray.push(props.client.query(usersQuery).toPromise())
+        }
+        let allUsersPromise = await Promise.all(usersPromiseArray)
+        
+        let users = []
+        for (var j=0; j<allUsersPromise.length; j++){
+            users.push(allUsersPromise[j].data.users)
+        }
```

Comment:
Good use case for Array.map

Also note that you should use let or const over var. That will give you block scope and you can always use "i" in for loops for example without worrying about namespace clash in scope.

```suggestion
        let users = allUsersPromise.map(userResolve => userResolve.data.users)
```